### PR TITLE
fix(checkbox): Fixed hover focus colors for unchecked checkbox

### PR DESCRIPTION
**Pull request by abhiomkar**
### Description

* Updated `mdc-checkbox-focus-indicator` mixin to allow customizing ripple color for checked checkboxes. Previously `mdc-checkbox-focus-indicator` use to only set color for JS disabled checkboxes.
* Included `mdc-checkbox-focus-indicator` mixin to `.mdc-checkbox` class name since this mixin uses state selectors such as `--selected` with root class.
* Updated screenshot files to include `--selected` class name for checked checkboxes.

Refs #2949

### CODE REVIEW

File: test/screenshot/spec/mdc-checkbox/fixture.scss [hunk 1]
```diff
@@ -32,7 +32,7 @@ $custom-checkbox-color: $material-color-red-300;
 
 .custom-checkbox--container-colors {
   @include mdc-checkbox-container-colors(
-    $unmarked-stroke-color: $custom-checkbox-color,
+    // $unmarked-stroke-color: $custom-checkbox-color,
```
Comment (moog16) [inline test/screenshot/spec/mdc-checkbox/fixture.scss]: was this not being used before? Should we just remove this.

Comment (abhiomkar) [reply test/screenshot/spec/mdc-checkbox/fixture.scss]: This is used in mixins/container-colors.html screenshot test file, removed the commented code.

Comment (moog16) [reply test/screenshot/spec/mdc-checkbox/fixture.scss]: Cool sounds good

**Comment by abhiomkar** [comment]
Ready for re-review.

**Pull request merged**